### PR TITLE
Redesign failover standby API

### PR DESCRIPTION
**Pull request by robertbaldyga**
Signed-off-by: Robert Baldyga <robert.baldyga@intel.com>

### CODE REVIEW

**Review by mmichal10 — APPROVED** [review]

File: src/mngt/ocf_mngt_cache.c [hunk 1]
```diff
@@ -2561,7 +2654,7 @@ void _ocf_mngt_cache_failover_detach(ocf_cache_t cache,
 }
 
 
-static void _ocf_mngt_cache_activate(ocf_cache_t cache,
+static void _ocf_mngt_cache_standby_activate(ocf_cache_t cache,
 		struct ocf_mngt_cache_device_config *device_cfg,
 		_ocf_mngt_cache_attach_end_t cmpl, void *priv1, void *priv2)
```
Comment (arutk) [inline src/mngt/ocf_mngt_cache.c:2659]: Please add "open_cores" parameter, similarly to cache load - this improves testability in pyocf

Comment (robertbaldyga) [reply src/mngt/ocf_mngt_cache.c:2659]: Done.

**Review by mmichal10 — APPROVED** [review]

**Review by mmichal10 — DISMISSED** [review]

File: src/mngt/ocf_mngt_cache.c [hunk 2]
```diff
@@ -2052,60 +2043,70 @@ static void _ocf_mngt_standby_init_properties(ocf_pipeline_t pipeline,
...
+
 	result = ocf_pio_concurrency_init(&cache->standby.concurrency, cache);
 	if (!result)
 		context->flags.pio_concurrency = true;
 
 	OCF_PL_NEXT_ON_SUCCESS_RET(context->pipeline, result);
 }
 
-static void _ocf_mngt_bind_recovery_unsafe(ocf_pipeline_t pipeline,
+static void _ocf_mngt_standby_recovery_unsafe(ocf_pipeline_t pipeline,
```
Comment (arutk) [inline src/mngt/ocf_mngt_cache.c:2097]: I think all the "unsafe" variants of recovery functions in this file are unnecessary - we do not need to handle invalid status bits combinations or stale mappings

Comment (arutk) [reply src/mngt/ocf_mngt_cache.c:2097]: I mean __populate_free_unsafe and _ocf_mngt_rebuild_metadata(initialized = false)

Comment (robertbaldyga) [reply src/mngt/ocf_mngt_cache.c]: Done.

**Review by arutk — APPROVED** [review]

**Review by rafalste — APPROVED** [review]

**Pull request merged**